### ISSUE TITLE
FileUpload: Upload broken for mode='advanced' and customUpload=true

### ISSUE THREAD

**Issue by mathiasengelhardt** [primefaces/primeng#15117]
### Describe the bug

When using FileUpload like this: 
`<p-fileUpload (uploadHandler)="handleFileUpload($event)" [auto]="true" [customUpload]="true"></p-fileUpload>`, the `uploadHandler` callback won't be called.
See fileupload.ts, line 563:
```
        if (this.hasFiles() && this.auto && !(this.mode === 'advanced') && !this.isFileLimitExceeded()) {
            this.upload();
        }
```

Prior to version 17.8.0, the (working) code was:
```
        if (this.hasFiles() && this.auto && (!(this.mode === 'advanced') || !this.isFileLimitExceeded())) {
            this.upload();
        }
```

### Environment

Windows

### Angular version

17.3.0

### PrimeNG version

17.11.0

### Build / Runtime

Angular CLI App

### Language

TypeScript

### Node version (for AoT issues node --version)

18.16.0

**Comment by rafparedis** [comment]
I also have this same problem. FileUpload component stopped working 

<p-fileUpload name="uploadedFiles[]" customUpload="true" auto="true" (uploadHandler)="onUpload(section.section,$event)"

also using customUpload

**Comment by astranCZ** [comment]
Even advanced mode with the auto attribute does not work

https://stackblitz.com/edit/gunrgk?file=src%2Fapp%2Fdemo%2Ffile-upload-advanced-demo.html

**Comment by Sara-pixie** [comment]
seems like the issue can be solved by using mode=basic, but if you actually need the advanced mode like us there's no way around it.

**Comment by cetincakiroglu** [comment]
Hi, 
This issue is duplicate of https://github.com/primefaces/primeng/issues/15150
The related issue is fixed and added to the 17.13.0 milestone. Closing this one.
